src/librustc_mir/transform/inline.rs

```diff
@@ -687,7 +691,19 @@ impl<'a, 'tcx> MutVisitor<'tcx> for Integrator<'a, 'tcx> {
                 // Return pointer; update the place itself
                 *place = self.destination.clone();
             },
-            _ => self.super_place(place, _ctxt, _location)
+            _ => {
+                self.visit_place_base(&mut place.base, context, location);
+
+                let new_projection: Vec<_> = place.projection.iter().map(|elem|
```

Comment:
if none of the projections are `PlaceElem::Index`, we just uselessly cloned all of them, collected a vector and boxed it

Reply:
FTR: We are going to try with Cow here.